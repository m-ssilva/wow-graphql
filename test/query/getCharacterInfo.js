module.exports = `{
  getCharacterInfo(region: US, realm: azralon, character_name: "JohnDoe") {
    profile {
      name
      gender
      faction
      race
      character_class
      specialization
      guild
      level
      item_level
      achievement_points
      covenant
      covenant_renown
    }
    media {
      character_image_url
    }
    arena {
      bracket {
        type
        rating
        match_statistics {
          played
          won
          win_rate
        }
      }
    }
  }
}`