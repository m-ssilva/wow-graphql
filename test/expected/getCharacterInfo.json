{
  "data": {
    "getCharacterInfo": {
      "profile": {
        "name": "JohnDoe",
        "gender": "Male",
        "faction": "Horde",
        "race": "Blood Elf",
        "character_class": "Paladin",
        "specialization": "Holy",
        "guild": "A random guild",
        "level": "60",
        "item_level": "192",
        "achievement_points": "13100",
        "covenant": "Kyrian",
        "covenant_renown": "32"
      },
      "media": {
        "character_image_url": "https://render-us.worldofwarcraft.com/character/azralon/38/178631206-main.jpg"
      },
      "arena": {
        "bracket": [
          {
            "type": "ARENA_2v2",
            "rating": 1292,
            "match_statistics": {
              "played": 60,
              "won": 36,
              "win_rate": "60.00%"
            }
          },
          {
            "type": "ARENA_3v3",
            "rating": 1128,
            "match_statistics": {
              "played": 29,
              "won": 15,
              "win_rate": "51.72%"
            }
          }
        ]
      }
    }
  }
}